### PR TITLE
Improve browsercheck script

### PR DESCRIPTION
**Pull request by bhollis**
1. Automatically accept newer versions of the browsers we accept (e.g. if we accept the latest two versions of Chrome, which are 78 and 79, we'll still accept 80 and 81).
2. Link to a new [support wiki](https://github.com/DestinyItemManager/DIM/wiki/Supported-Browsers) about which browsers we support.
3. Remove the runtime dependency on browserslist, greatly reducing the size of this script. I realized I could just compute it as part of the build.
4. Force the script's dependencies to be bundled with it, so it doesn't split into multiple chunks. This helps it continue to run even if the rest of the webpack entry points fail.
5. Avoid running babel on browserscheck - it's meant to be written as maximally-compatible ES5 already and we don't want to inject core-js polyfills.
5. Remove "unreleased versions" which avoids [this bug in babel-preset-env](https://github.com/babel/babel/issues/10945).

I may just want to switch to https://github.com/ElemeFE/obsolete-webpack-plugin later, since it seems like it's just a better (if much more complicated) version of this, but for now these changes help.

### CODE REVIEW

**Comment by bhollis** [comment]
Well that's disappointing. I thought I was getting around the babel bug. We may have to just wait for the fix.

**Comment by sundevour** [comment]
i'm getting a new-to-me "Compiled with warnings." which i assume you know,
and in firefox 71 (current afaik) a not-supported warning
chrome seems fine

**Comment by sundevour** [comment]
nevermind the js UA is
`"Mozilla/5.0 (Windows NT 6.3; Win64; x64; rv:72.0) Gecko/20100101 Firefox/72.0"`

**Comment by bhollis** [comment]
Cool, I'll test it out. I don't know what the "Compiled with warnings" thing is - I'm not seeing any warnings in the CLI output.

**Comment by bhollis** [comment]
Er, I guess I am seeing the warnings, but I don't know what they are. The toaster says something about immer (which we recently upgraded) but I don't see the full output.

**Comment by sundevour** [comment]
same yeah it's just toasting :|

**Comment by bhollis** [comment]
```
WARNING in ./node_modules/immer/dist/immer.module.js
Module Warning (from ./node_modules/source-map-loader/index.js):
(Emitted value instead of an instance of Error) Cannot find source file '../node_modules/tslib/tslib.es6.js': Error: Can't resolve '../node_modules/tslib/tslib.es6.js' in '/Users/brh/Documents/oss/DIM/node_modules/immer/dist'
 @ ./src/app/item-review/reducer.ts 10:0-28 46:15-22 67:15-22 85:15-22
 @ ./src/Index.tsx
```

🤷‍♂

**Comment by bhollis** [comment]
https://github.com/immerjs/immer/issues/487

**Comment by bhollis** [comment]
OK I added some Jest tests - feel free to throw more user agent strings at me.

File: src/browsercheck.js [hunk 1]
```diff
@@ -20,14 +19,14 @@ function getBrowserName(agent) {
...
-  var browserName = getBrowserName(agent);
+  var browserName = getBrowserName(agent).toLowerCase();
   var version = (browserName === 'ios_saf'
     ? agent.os.version
     : agent.browser.version || agent.os.version || ''
   ).split('.');
   while (version.length > 0) {
     try {
-      return browserslist(browserName + ' ' + version.join('.'))[0];
+      return browserName + ' ' + version.join('.');
```
Comment (delphiactual) [inline src/browsercheck.js]:
```suggestion
      return `${browserName}  ${version.join('.')}`;
```

File: src/browsercheck.js [hunk 2]
```diff
@@ -36,24 +35,54 @@ function getBrowserVersionFromUserAgent(agent) {
...
+  }
+
+  var browser = getBrowserVersionFromUserAgent(agent);
+  var supported = isBrowserSupported(browser);
+
+  if (!supported && agent.os.name !== 'Android') {
+    // Detect anything based on chrome as if it were chrome
+    var chromeMatch = /Chrome\/(\d+)/.exec(agent.ua);
+    if (chromeMatch) {
+      browser = 'chrome ' + chromeMatch[1];
```
Comment (delphiactual) [inline src/browsercheck.js]:
```suggestion
      browser = `chrome ${chromeMatch[1]}`
```

File: src/browsercheck.js [hunk 3]
```diff
@@ -36,24 +35,54 @@ function getBrowserVersionFromUserAgent(agent) {
...
+      supported = isBrowserSupported(browser);
+    }
   }
+  if (!supported) {
+    console.warn(
+      'Browser ' + browser + ' is not supported by DIM. Supported browsers:',
```
Comment (delphiactual) [inline src/browsercheck.js]:
```suggestion
      `Browser ${browser} is not supported by DIM. Supported browsers:`,
```

File: src/browsercheck.js (hunk 1 above)
Comment (bhollis) [reply src/browsercheck.js]: Not for this file, which needs to stay es5-compatible.

**Pull request merged**